config/initializers/omniauth.rb

```diff
@@ -1,6 +1,15 @@
 Rails.application.config.middleware.use OmniAuth::Builder do
   ENV["GITHUB_KEY"] ||= "03fafb19e89f60f1f0f3"
   ENV["GITHUB_SECRET"] ||= "2fd993d61ff8eefa127ee5eee0507bdfa5312c44"
+
   provider :github, ENV["GITHUB_KEY"], ENV["GITHUB_SECRET"],
            scope: "user,repo"
 end
+
+unless Rails.env.production? || Rails.env.test?
+  OmniAuth::Strategies::OAuth2.class_eval do
```

Comment:
This is the cleanest way we could possibly change OmniAuth behavior to support "forwarding". It relies on OmniAuth internals, though, and it will break once we update the gem (assuming they change where this method lives).

As you can notice, we still have to include some `if` clauses which will be totally useless in production: however, I am glad that building the forwarder moved the problem to the configuration, so that we put such conditionals only when initializing the application (more or less).